pkgs/development/compilers/ghc/common-hadrian.nix

```diff
@@ -208,6 +217,10 @@
            then ./Cabal-at-least-3.6-paths-fix-cycle-aarch64-darwin.patch
            else ./Cabal-3.12-paths-fix-cycle-aarch64-darwin.patch)
         ]
+        ++ lib.optionals stdenv.targetPlatform.isWindows [
+          # https://gitlab.haskell.org/ghc/ghc/-/issues/23688
+          ./win32-no-hsc2hs.patch
```

Comment:
I'm not sure whether this is really the way to go about it, but it does make Win32 build

Reply:
Not only that, I managed to delete it during pre-commit cleanup. Derp.

Reply:
The linked issue only mentions tests as a problem, we don't build nor run the GHC tests…

Reply:
Well, yes, and the error message in the issue isn't what I had either. I simply got the usual "missing dependency" regarding hsc2hs while building Win32.

Maybe I should instead link directly to the comment in that issue?

> The problem seems to be that Hadrian doesn't take into account the build-tool-depends stanza of Win32's cabal file.

I guess one alternative would be opening a separate issue and/or PR